propagation/baggage_test.go

```diff
@@ -113,13 +113,79 @@ func TestExtractValidBaggageFromHTTPReq(t *testing.T) {
 				{Key: "key1", Value: "val%2"},
 			},
 		},
+		{
+			name:   "empty header",
+			header: "",
+			want:   members{},
+		},
 	}
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
+			mapCarr := propagation.MapCarrier{}
+			mapCarr["baggage"] = tt.header
 			req, _ := http.NewRequest(http.MethodGet, "http://example.com", nil)
 			req.Header.Set("baggage", tt.header)
 
+			// test with http header carrier (which implements ValuesGetter)
+			ctx := prop.Extract(context.Background(), propagation.HeaderCarrier(req.Header))
+			expected := tt.want.Baggage(t)
+			assert.Equal(t, expected, baggage.FromContext(ctx))
```

Comment:
It would be nicer to have subtests for MapCarrier and HeaderCarrier, but I can also address it myself in a separate PR. 

For now this would be good enough for me:

```suggestion
			assert.Equal(t, expected, baggage.FromContext(ctx), "should extract baggage for HeaderCarrier")
```